### PR TITLE
libobs: Add bitstream reader with unit tests

### PR DESCRIPTION
**Pull request by DDRBoxman**
### Description

Adds cmocka a C unit testing framework to use with libobs. Also adds a bitstream reader for another feature I'm working on.

### Motivation and Context

Unit tests are cool. I have a branch that adds captions over SDI that needs a bitstream reader. Bitstream readers are cool. This bitstream reader has unit tests.

### How Has This Been Tested?

Unit tests :)

### Types of changes

Add bitstream reader and unit tests for libobs

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested. 😄 
- [x] All commit messages are properly formatted and commits squashed where appropriate.

### CODE REVIEW

**Comment by DDRBoxman** [comment]
Adding tests to windows CI is pretty simple, but we would need to add cmocka to the deps zip first

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -2,8 +2,6 @@ name: 'CI Multiplatform Build'
 
 on:
   push:
-    branches:
-      - master
```
Comment (RytoEX) [inline .github/workflows/main.yml:6]: Why was this removed?

File: CI/scripts/macos/Brewfile [hunk 2]
```diff
@@ -6,3 +6,4 @@ brew "fdk-aac"
 brew "https://gist.githubusercontent.com/DDRBoxman/9c7a2b08933166f4b61ed9a44b242609/raw/ef4de6c587c6bd7f50210eccd5bd51ff08e6de13/qt.rb"
 brew "swig", link: false
 brew "https://gist.githubusercontent.com/DDRBoxman/4cada55c51803a2f963fa40ce55c9d3e/raw/572c67e908bfbc1bcb8c476ea77ea3935133f5b5/swig.rb"
+brew "cmocka"
```
Comment (RytoEX) [inline CI/scripts/macos/Brewfile:9]: Add a newline.

File: libobs/util/bitstream.c [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+{
+	memset(r, 0, sizeof(struct bitstream_reader));
+	r->buf = data;
+	r->subPos = 0x80;
+	r->len = len;
+}
+
+uint8_t bitstream_reader_read_bit(struct bitstream_reader *r)
+{
+	if (r->pos >= r->len) {
```
Comment (RytoEX) [inline libobs/util/bitstream.c:17]: No need for braces on a single statement `if`.

File: libobs/util/bitstream.c [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+uint8_t bitstream_reader_r8(struct bitstream_reader *r)
+{
+	return bitstream_reader_read_bits(r, 8);
+}
+
+uint16_t bitstream_reader_r16(struct bitstream_reader *r)
+{
+	uint8_t b = bitstream_reader_read_bits(r, 8);
+	return ((uint16_t)b << 8) | bitstream_reader_read_bits(r, 8);
+}
```
Comment (RytoEX) [inline libobs/util/bitstream.c:53]: Add a newline.

File: libobs/util/bitstream.h [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+	uint8_t subPos;
+	uint8_t *buf;
+	size_t len;
+};
+
+void bitstream_reader_init(struct bitstream_reader *r, uint8_t *data,
+			   size_t len);
+uint8_t bitstream_reader_read_bits(struct bitstream_reader *r, int bits);
+uint8_t bitstream_reader_r8(struct bitstream_reader *r);
+uint16_t bitstream_reader_r16(struct bitstream_reader *r);
```
Comment (RytoEX) [inline libobs/util/bitstream.h:16]: Add a newline.

File: test/CMakeLists.txt [hunk 6]
```diff
@@ -1,10 +1,16 @@
...
-if(APPLE AND UNIX)
-	add_subdirectory(osx)
+	if(APPLE AND UNIX)
+		add_subdirectory(osx)
+	endif()
 endif()
+
+if (ENABLE_UNIT_TESTS)
+	add_subdirectory(cmocka)
+endif()
```
Comment (RytoEX) [inline test/CMakeLists.txt:16]: Add a newline.

File: test/cmocka/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+add_test(test_serializer ${CMAKE_CURRENT_BINARY_DIR}/test_serializer)
+fixLink(test_serializer)
+
+
+# Bitstream test
+add_executable(test_bitstream test_bitstream.c)
+target_link_libraries(test_bitstream ${CMOCKA_LIBRARIES} libobs)
+
+add_test(test_bitstream ${CMAKE_CURRENT_BINARY_DIR}/test_bitstream)
+fixLink(test_bitstream)
```
Comment (RytoEX) [inline test/cmocka/CMakeLists.txt:39]: Add a newline.

File: test/cmocka/test_bitstream.c [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+int main()
+{
+	const struct CMUnitTest tests[] = {
+		cmocka_unit_test(bitstream_test),
+	};
+
+	return cmocka_run_group_tests(tests, NULL, NULL);
+}
```
Comment (RytoEX) [inline test/cmocka/test_bitstream.c:34]: Add a newline.

File: test/cmocka/test_serializer.c [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+int main()
+{
+	const struct CMUnitTest tests[] = {
+		cmocka_unit_test(serialize_test),
+	};
+
+	return cmocka_run_group_tests(tests, NULL, NULL);
+}
```
Comment (RytoEX) [inline test/cmocka/test_serializer.c:31]: Add a newline.

File: .github/workflows/main.yml (hunk 1 above)
Comment (DDRBoxman) [reply .github/workflows/main.yml:6]: This makes the CI process run on any branch of the obs repo, right now it only runs on master.

**Comment by notr1ch** [comment]
I would prefer to see this committed with the feature that requires it, right now this seems to be adding unnecessary / unused functions to libobs.
